File: src/client/src/app/models/project.ts
import { DataTableRow } from ".";
import { Guid } from "guid-typescript";
import { Enterprise } from "./enterprise";

export interface Project {
    id: Guid
    name: string;
    startingInvestmentSum: number;
    enterprise: Enterprise;
    enterpriseId: Guid;
}

export interface ProjectRow extends DataTableRow {
    id: string,
    internalId: string;
    name: string;
    startingInvestmentSum: number;
    enterprise: string;
    enterpriseId: string;
}

File: src/server/InvestmentApp-Server/Models/Project.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace InvestmentApp.Models;

public class Project
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public string Name { get; set; }

    public double StartingInvestmentSum { get; set; }

    public Enterprise Enterprise { get; set; }

    public Guid EnterpriseId { get; set; }
}
